src/Containers/NutritionFact/NutritionFact.tsx

```diff
@@ -0,0 +1,296 @@
+            return <div>{secondLabelState}</div>
+        }
+        return null
+    },[secondLabelState])
+
+    return (
+        <EntryContainer separatorWidth={separatorWidth} {...props} ref={ref}>
+            <div>{label||delayedLabel}</div>
+            {renderSecondLabel()}
+        </EntryContainer>
+    )
+})
+
+interface IEntryProps extends ICommonEntryProps{
+    amount?:number;
+    dailyAmount:number;
+    label:string;
+    unity:string;
+    indentationNumber?:number;
+    indentationSize?:number;
+}
+
+const Entry:React.FC<IEntryProps>=({
+    amount=0,
+    dailyAmount,
+    label,
+    unity,
+    editMode,
+    bold=false,
+    separatorWidth=1,
+    indentationNumber=0,
+    indentationSize=4,
+    ...props
+}):React.ReactElement=>{
+    const [amountState,setAmountState]=useState<string>(amount.toString())
+    
+    /**
+     * this computes the daily value percentage
+     */
+    const dailyValuePercentage= useMemo(()=>{
+        if(dailyAmount){
+            /**
+             * gets the percentage of a value respect a totalValue
+             * @param value {number} - value for which we want to get percentage
+             * @param totalValue {number} - value for which percentage is 100 
+             * @returns {number} the percentage
+             */
+            const getPercentage=(value:number,totalValue:number)=>Math.round(value/totalValue*100)
+            return getPercentage(parseInt(amountState,10),dailyAmount) 
+        }
+        return null
+    },[dailyAmount,amountState]);
+
+    /**
+     * computes the amount of white space
+     * @returns {string} white space 
+     */
+    const renderIndentation=useCallback(()=>{
+        let space=''
+        const nbsp='\u00a0'
+        for(let j=0;j<indentationNumber;j+=1){
+            for(let i=0;i<indentationSize; i+=1) {
+                space+=nbsp
+            }
+        }
+        return space
+    },[indentationNumber,indentationSize])
+
+    const updateAmountState=({target}:React.ChangeEvent<HTMLInputElement>)=>{
+        setAmountState(target.value)
+    }
+
+    /**
+     * renders the amount depending on editMode
+     */
+    const renderAmount=useCallback(()=>{
+        if(editMode){
+            return <Input onChange={updateAmountState} value={amountState} width={ENTRY_INPUT_WIDTH} />
+        }
+        return amountState
+    },[editMode,amountState])
+
+    return (
+        <EntryContainer separatorWidth={separatorWidth} justifyContent='space-between' {...props}>
+            <LabelContainer> 
+                {renderIndentation()}
+                <Bold bold={bold}>
+                    {label}
+                    &nbsp;
+                </Bold>
+                <AmountContainer>
+                    {renderAmount()}
+                    {unity}
+                </AmountContainer>
+            </LabelContainer>
+            {dailyValuePercentage!==null&&(
```

Comment:
Cant you use
!dailyValuePercentage instead? What is the value is undefined...

Reply:
!dailyValuePercentage don't work because we want also to represent (display) values 0. dailyValuePercentage could not be undefined because it's of type number|null

Reply:
Then use ?? Instead.

Reply:
no, ?? doesn't serve because someVar??expresion executes expresion if someVar is null, and I don't want this, I want the contrary